src/Microsoft.Health.Fhir.SqlServer/Features/Schema/ResourceVersion.cs

```diff
@@ -0,0 +1,32 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Data.SqlClient;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Schema
+{
+    public static class ResourceVersion
+    {
+        private static bool versionIsLong = true;
+
+        public static int GetVersion(SqlDataReader reader, int index)
+        {
+            int version = 0;
+            retryVersion:
+            try
+            {
+                version = versionIsLong ? (int)reader.GetInt64(index) : reader.GetInt32(index);
```

Comment:
Is there a way to get the type from the SqlDataReader instead?

Reply:
SqlDataReader uses the type defined in the model by default. Type defined in the model and DB should be same unless we have a code rollback where Model has type as Int32 and DB has type as Int64. This code is to take care of compatibility issue in case of code rollback.

Reply:
SqlDataReader is returned by ExecuteReaderAsync().  Where does it get the model from?

Reply:
I don't think `(int)reader.GetInt64(index)` will be compatible if it actually does return a long since it won't fit into an int.

Reply:
@cunninghamjc 
When we read, we do pass column type. See example below. That is how reader gets the type information.
reader.Read(VLatest.Resource.ResourceId, 1);

Reply:
@brendankowitz 
you are right, (int)reader.GetInt64(index) won't be compatible if version actually have a long value. But we are not going to allow customers to send version as long until we are done with iteration 2.
This code is in place to handle rollback situation. Even though column type is long, existing values can fit it into int. This code will be cleaned up in iteration 3 to read it as long.

Reply:
In iteration 2, what happens if an upgraded instance of a FHIR service accepts a long and writes it to the DB, then a non-upgraded instance of the FHIR service reads it back out?

Reply:
@cunninghamjc @brendankowitz code is modified to check the type from SqlDataReader. Please take a look.
@cunninghamjc schema update starts after deploying applications. So there will not be a scenario few instances are on new bits and few are on old.
But there is a scenario where after releasing iteration 2 and consuming long version, if we rollback the code, we will run into issue. To avoid that issue,
We will not advertise this new change until we feel comfortable.
And also, we will try avoiding having several major changes going in the same release as iteration2 to minimize reasons for rollback.